### ISSUE TITLE
Shaping regression in Mongolian final forms

### ISSUE THREAD

**Issue by simoncozens** [notofonts/mongolian#23]
As noted by @lianghai, there was a shaping regression somewhere between 1.03 and 2.01 in the handling of Mongolian final forms.

Here's 2.000, googlefonts/noto-source@94ac96305, which is the earliest version in noto-source:
```
$ shape master_ttf/NotoSansMongolian-Regular.ttf -u 1825,1837,182f,1825,182d
[u1825.ini=0+1360|u1837.med=1+559|u182F.med=2+284|u1825.med=3+573|u182D.fin=4+663]
```
![shape](https://user-images.githubusercontent.com/106728/195975149-9cdd138d-da66-4a80-b22b-0bedc27d1360.png)

Here's 1.003:

```
[gid177=0+2785|gid31=1+1145|gid75=2+582|gid943=3+1174|gid956=4+1368]
```

![shape](https://user-images.githubusercontent.com/106728/195975210-060e5885-c93d-4020-8670-09039f2119de.png)

My suspicion is that the regression was introduced while attempting to deal with #12, but I don't have access to earlier sources.

Note that Mongolian shaping is *still* under discussion by Unicode, so this will require some research to fix properly.

**Comment by lianghai** [comment]
Constructed an example to show the four known major mistakes (the feminine coda “g” final is already shown in the OP), reported by @shenyileirob and @Kushim-Jiang:

1. Feminine (C)VC syllables’ coda “g”, medial and final.
2. (C)V(C) syllables’ onset and coda “d”, medial.

Input characters, in constructed pseudo words “egneg edde” to demonstrate the contexts:

- 1821 ᠡ MONGOLIAN LETTER E
- 182D ᠭ MONGOLIAN LETTER GA
- 1828 ᠨ MONGOLIAN LETTER NA
- 1821 ᠡ MONGOLIAN LETTER E
- 182D ᠭ MONGOLIAN LETTER GA
- 0020 SPACE
- 1821 ᠡ MONGOLIAN LETTER E
- 1833 ᠳ MONGOLIAN LETTER DA
- 1833 ᠳ MONGOLIAN LETTER DA
- 1821 ᠡ MONGOLIAN LETTER E

<img width="552" alt="Screen Shot 2022-10-15 at 14 24 23" src="https://user-images.githubusercontent.com/343259/195986227-c1edf068-acfd-49cf-bf66-27bd156bb12f.png">

1.040 glyphs (correct):

- e: u1821.ini
- g: u182D.medV2
- n: u1828.med
- e: u1821.med
- g: u182D.finV2
- space
- e: u1821.ini
- d: u1833.medV1
- d: u1833.med
- e: u1821.fin

2.000 glyphs (regressions in bold, note many have the same glyph names but wrong shapes):

- e: u1821.ini
- g: **u182D.medV2**
- n: u1828.med
- e: u1821.med
- g: **u182D.fin**
- space
- e: u1821.ini
- d: **u1833.medV1**
- d: **u1833.med**
- e: u1821.fin

**Comment by lianghai** [comment]
Again, the Mongolian shaping model has been under debate, but how these four basic cases shown above should shape has never been controversial.

**Comment by simoncozens** [comment]
Fixed in #33. Will be released this quarter.
